dev/unix/volta-install.sh

```diff
@@ -0,0 +1,584 @@
+    request "       Please delete $(legacy_dir) and try again."
+    eprintf ""
+    eprintf "(We plan to implement automatic upgrades in the future. Thanks for bearing with us!)"
+    eprintf ""
+    return 1
+  fi
+  return 0
+}
+
+# Check if it is OK to upgrade to the new version
+upgrade_is_ok() {
+  local will_install_version="$1"
+  local install_dir="$2"
+
+  local volta_bin="$install_dir/volta"
+
+  # this is not able to install Volta prior to 0.5.0 (when it was renamed)
+  if [[ "$will_install_version" =~ ^([0-9]+\.[0-9]+) ]]; then
+    local major_minor="${BASH_REMATCH[1]}"
+    case "$major_minor" in
+      0.1|0.2|0.3|0.4)
+        eprintf ""
+        error "Cannot install Volta prior to version 0.5.0 (when it was named Notion)"
+        request "    To install Notion version $will_install_version, please check out the source and build manually."
+        eprintf ""
+        return 1
+        ;;
+    esac
+  fi
+
+  if [[ -n "$install_dir" && -x "$volta_bin" ]]; then
+    local prev_version="$( ($volta_bin --version 2>/dev/null || echo 0.1) | sed -E 's/^.*([0-9]+\.[0-9]+\.[0-9]+).*$/\1/')"
+    # if installing the same version, this is a no-op
+    if [ "$prev_version" == "$will_install_version" ]; then
+      eprintf "Version $will_install_version already installed"
+      return 1
+    fi
+    # in the future, check $prev_version for incompatible upgrades
+  fi
+  return 0
+}
+
+# returns the os name to be used in the packaged release,
+# including the openssl info if necessary
+parse_os_info() {
+  local uname_str="$1"
+  local openssl_version="$2"
+
+  case "$uname_str" in
+    Linux)
+      parsed_version="$(parse_openssl_version "$openssl_version")"
+      exit_code="$?"
+      if [ "$exit_code" != 0 ]; then
+        return "$exit_code"
+      fi
+
+      echo "linux-openssl-$parsed_version"
+      ;;
+    Darwin)
+      echo "macos"
+      ;;
+    *)
+      return 1
+  esac
+  return 0
+}
+
+parse_os_pretty() {
+  local uname_str="$1"
+
+  case "$uname_str" in
+    Linux)
+      echo "Linux"
+      ;;
+    Darwin)
+      echo "macOS"
+      ;;
+    *)
+      echo "$uname_str"
+  esac
+}
+
+# return true(0) if the element is contained in the input arguments
+# called like:
+#  if element_in "foo" "${array[@]}"; then ...
+element_in() {
+  local match="$1";
+  shift
+
+  local element;
+  # loop over the input arguments and return when a match is found
+  for element in "$@"; do
+    [ "$element" == "$match" ] && return 0
+  done
+  return 1
+}
+
+# parse the OpenSSL version from the input text
+parse_openssl_version() {
+  local version_str="$1"
+
+  # array containing the SSL libraries that are supported
+  # would be nice to use a bash 4.x associative array, but bash 3.x is the default on OSX
+  SUPPORTED_SSL_LIBS=( 'OpenSSL' )
+
+  # use regex to get the library name and version
+  # typical version string looks like 'OpenSSL 1.0.1e-fips 11 Feb 2013'
+  if [[ "$version_str" =~ ^([^\ ]*)\ ([0-9]+\.[0-9]+\.[0-9]+) ]]
```

Comment:
In a complete reversal of my previous comment, it turns out we actually don't care about the patch version, and only care about the major and minor, so we should just match to those (sorry).

Reply:
Ah, right! I saw that change in #409 and forgot to change it in this PR - will update that to match the current behavior of `boot-install.sh`